tm2/pkg/sdk/auth/genesis.go

```diff
@@ -0,0 +1,58 @@
+package auth
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+)
+
+// GenesisState - all state that must be provided at genesis
+type GenesisState struct {
+	Params Params `json:"params" yaml:"params"`
+}
+
+// NewGenesisState - Create a new genesis state
+func NewGenesisState(params Params) GenesisState {
+	return GenesisState{params}
+}
+
+// DefaultGenesisState - Return a default genesis state
+func DefaultGenesisState() GenesisState {
+	return NewGenesisState(DefaultParams())
+}
+
+// ValidateGenesis performs basic validation of genesis data returning an
+// error for any failed validation criteria.
+func ValidateGenesis(data GenesisState) error {
+	return data.Params.Validate()
+}
+
+// InitGenesis - Init store state from genesis data
+func (ak AccountKeeper) InitGenesis(ctx sdk.Context, data GenesisState) {
+	if amino.DeepEqual(data, GenesisState{}) {
+		return
+	}
+
+	if err := ValidateGenesis(data); err != nil {
+		panic(err)
```

Comment:
We can return errors, and handle them

Reply:
I think it should panic and stop loading an invalid genesis to the application.